app/models/poll.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+class Poll < ActiveRecord::Base
+  has_many :options, dependent: :destroy
+
+  validates :title, presence: true
+
+  after_touch :ensure_status_is_correct
+
+  def make_ready
```

Comment:
Additional timestamp columns?
Like `closed_at` is not null for closed polls?
Then custom scopes (`scope :published, -> { where.not(published_at: nil).where(finished_at: nil) }`) will allow you to select the desired subset of polls with ease.